src/slicers/card.js

```diff
@@ -21,8 +22,41 @@ const cardSlice = createSlice({
       const { id, text } = action.payload;
       state[id].text = text;
     },
+    updateCardColumnID(state, action) {
+      const { id, columnID } = action.payload;
+      state[id].columnID = columnID;
+    },
+    updateCardsIndexOnDragEnd(state, action) {
+      const { id, index, columnID } = action.payload;
+      state[id].index = index;
+      Object.values(state)
+        .filter((card) => card.columnID === columnID)
+        .forEach((card) => {
+          if (card.index >= index && card.id !== id) {
+            card.index = card.index + 1;
+          }
+        });
+    },
+    updateCardsIndexOnDragStart(state, action) {
+      const { id } = action.payload;
+      const { columnID, index } = state[id];
+      Object.values(state)
+        .filter((card) => card.columnID === columnID)
+        .forEach((card) => {
+          if (card.index > index) {
+            card.index = card.index - 1;
+          }
+        });
+    },
```

Comment:
This are really cluttered. Add some space, put things into variables with descriptive names. The code is for the machine mainly, but also for human, readability is a top priority.
```suggestion
      const { id } = action.payload;
      const { columnID, index } = state[id];

      const values = Object.values(state);
      const uniqueValues = values.filter((card) => card.columnID === columnID);
      uniqueValues.forEach((card) => {
          const isCardUnderMovedCard = card.index > index;

          if (isCardUnderMovedCard) {
            card.index = card.index - 1;
          }
        });
    },
```
There is still some room for improvement but something like that at least, in my opinion, is more readable, what your thought on this?